pyro/distributions/multivariate_normal.py

```diff
@@ -29,15 +29,30 @@ def linear_solve_compat(matrix, matrix_chol, y):
 
 def matrix_inverse_compat(matrix, matrix_chol):
     """Computes the inverse of a positive semidefinite square matrix."""
-    assert matrix.requires_grad == matrix_chol.requires_grad
-    if matrix.requires_grad:
+    if matrix.requires_grad or matrix_chol.requires_grad:
         # If derivatives are required, use the more expensive inverse.
         return torch.inverse(matrix)
     else:
         # Use the cheaper Cholesky based potri.
         return potri_compat(matrix_chol)
 
 
+class _NormalizationConstant(Function):
```

Comment:
nit: It would be nice to have a one-line comment explaining the purpose of this class, something like:
> This computes either zero or the true normalization constant depending on `normalized`, but always computes the true tradient.